backend/src/test/controller/PromotionController.test.ts

```diff
@@ -74,6 +83,17 @@ describe('Unit tests for PromotionController', function () {
     await promotionRepository.save(promotion1);
     await promotionRepository.save(promotion2);
 
+    await mockRedisClient.setex(
+      promotion1.placeId,
+      2592000,
+      JSON.stringify({ lat: promotion1.lat, lon: promotion1.lon })
+    );
+    await mockRedisClient.setex(
+      promotion2.placeId,
+      2592000,
+      JSON.stringify({ lat: promotion2.lat, lon: promotion2.lon })
+    );
```

Comment:
I think there is still a lot of duplicated code for this test class.

I was thinking something like this
```
await setLatLon([promotion1, promotion2])
```

and you define the function `setLatLon`. What do you think?